### PR TITLE
gh-125168: Update __future__.rst typo

### PR DESCRIPTION
**Pull request by JamesMcCarthy21**

### CODE REVIEW

File: Doc/library/__future__.rst [hunk 1]
```diff
@@ -66,7 +66,7 @@ language using this mechanism:
 +------------------+-------------+--------------+---------------------------------------------+
 | annotations      | 3.7.0b1     | Never [1]_   | :pep:`563`:                                 |
 |                  |             |              | *Postponed evaluation of annotations*,      |
-|                  |             |              | :pep:`649`: *Deferred evalutation of        |
+|                  |             |              | :pep:`649`: *Deferred evaluation of        |
```
Comment (skirpichev) [inline Doc/library/__future__.rst]: Try to fix sphinx formatting: add extra space after "evaluation of".  I think it should fix CI failures.

See https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#rst-tables

Comment (JamesMcCarthy21) [reply Doc/library/__future__.rst]: Ok, thank you for the help

Comment (skirpichev) [reply Doc/library/__future__.rst]: Next time you might want to look on https://devguide.python.org/

E.g. documentation-related stuff: https://devguide.python.org/documentation/ (including rst markup, howto build docs, etc)

**Review by skirpichev — APPROVED** [review]

**Review by Eclips4 — APPROVED** [review]
Thanks!

**Pull request merged**